client/components/platform-checkout/save-user/checkout-page-save-user.js

```diff
@@ -14,18 +14,50 @@ import AdditionalInformation from './additional-information';
 import PhoneNumberInput from './phone-number-input';
 import Agreement from './agreement';
 import { getConfig } from 'utils/checkout';
-import './style.scss';
 
 const CheckoutPageSaveUser = () => {
 	const [ isSaveDetailsChecked, setIsSaveDetailsChecked ] = useState( false );
 	// eslint-disable-next-line no-unused-vars
 	const [ phoneNumber, setPhoneNumber ] = useState( '' );
+	const [ isPhoneValid, onPhoneValidationChange ] = useState( null );
 	const isRegisteredUser = usePlatformCheckoutUser();
 	const {
 		isWCPayChosen,
 		isNewPaymentTokenChosen,
 	} = useSelectedPaymentMethod();
 
+	// This runs once to populate the phone number state if already present in the DOM.
+	useEffect( () => {
+		setPhoneNumber(
+			document.getElementById( 'billing_phone' )?.value ?? ''
+		);
+	}, [] );
```

Comment:
Do you mean the phone input field is not populated and we can get rid of this, or this gets fired before the population of the field? 🤔

Reply:
>  or this gets fired before the population of the field? 

Yes, this statement is right.

So now it's fired on the checkout page load only. Previously it used to fire on mount the phone input component. If I check/uncheck the `save user` checkbox the phone input component mounts/unmounts. 

Consider this flow on trunk-
- checkout page loads and the phone field (the one in the billing address section) is empty initially. The checkbox is also unchecked
- User adds a phone number
- User selects the checkbox
- The phone number input is populated. This happens because the phone number input component mounts on selecting the checkbox, the hook is triggered and takes the value of the `billing_phone`.

Now in this branch-
- checkout page loads and the phone field (the one in the billing address section) is empty initially. The checkbox is also unchecked. But on the checkbox component the hook is triggered and received empty value as `billing_phone`
- User adds a phone number
- User selects the checkbox
- The phone number input field is empty now because the hook does not get triggered on this child component mount and shows the empty string that was initially obtained from the `billing_phone` value on it's parent component.

So the behavior is deviated from how it is on the trunk. Does this make sense? :thinking:

Reply:
Thank you for the detailed steps, I can reproduce it now and I think it's fixed by https://github.com/Automattic/woocommerce-payments/pull/4355/commits/9b934c8939f9e893b96441dd9e72b9799e1cec08